test/modelmatrix.jl

```diff
@@ -402,5 +402,35 @@
         f = apply_schema(@formula(0 ~ a&b&c), schema(t))
         @test vec(modelcols(f.rhs, t)) == modelcols.(Ref(f.rhs), Tables.rowtable(t))
     end
+
+    @testset "#293 conversion of Float32s" begin
+        d = (; y=rand(Float32, 4), x=rand(Float32, 4), z=[1:4; ])
+        f = @formula(y ~ 1 + x + log(x) * z)
+        y, x = modelcols(apply_schema(f, schema(d)), d)
+        @test eltype(y) == eltype(x) == Float32
+
+        f0 = @formula(y ~ 0 + x + log(x) * z)
+        y, x = modelcols(apply_schema(f0, schema(d)), d)
+        @test eltype(y) == eltype(x) == Float32
+
+        fint = @formula(y ~ 1 + z)
+        y, x = modelcols(apply_schema(fint, schema(d)), d)
+        @test eltype(x) == Int
+
+        contr = DummyCoding(; base=4, levels=1:4)
+        d = (; y=rand(Float32, 4), x=rand(Float32, 4), z=[1:4; ])
```

Comment:
this isn't yet used below, right?

Reply:
it is used but it's redundant with the definition above...

Reply:
oh wait the contr part?  yeah that's right.